runtime/lua/vim/lsp/log.lua

```diff
@@ -12,130 +12,124 @@ log.levels = vim.deepcopy(vim.log.levels)
+    if levelnr < current_log_level then
       return false
     end
-
-    local log_info = vim.uv.fs_stat(logfilename)
-    if log_info and log_info.size > 1e9 then
-      local warn_msg = string.format(
-        'LSP client log is large (%d MB): %s',
-        log_info.size / (1000 * 1000),
-        logfilename
-      )
-      notify(warn_msg)
+    local argc = select('#', ...)
+    if argc == 0 then
+      return true
     end
-
-    -- Start message for logging
-    logfile:write(string.format('[START][%s] LSP logging initiated\n', os.date(log_date_format)))
-    return true
-  end
-
-  for level, levelnr in pairs(log.levels) do
-    -- Also export the log level on the root object.
-    log[level] = levelnr
-    -- FIXME: DOC
-    -- Should be exposed in the vim docs.
-    --
-    -- Set the lowercase name as the main use function.
-    -- If called without arguments, it will check whether the log level is
-    -- greater than or equal to this one. When called with arguments, it will
-    -- log at that level (if applicable, it is checked either way).
-    --
-    -- Recommended usage:
-    -- ```
-    -- if log.warn() then
-    --   log.warn("123")
-    -- end
-    -- ```
-    --
-    -- This way you can avoid string allocations if the log level isn't high enough.
-    if level ~= 'OFF' then
-      log[level:lower()] = function(...)
-        local argc = select('#', ...)
-        if levelnr < current_log_level then
-          return false
-        end
-        if argc == 0 then
-          return true
-        end
-        if not open_logfile() then
-          return false
-        end
-        local info = debug.getinfo(2, 'Sl')
-        local header = string.format(
-          '[%s][%s] ...%s:%s',
-          level,
-          os.date(log_date_format),
-          string.sub(info.short_src, #info.short_src - 15),
-          info.currentline
-        )
-        local parts = { header }
-        for i = 1, argc do
-          local arg = select(i, ...)
-          if arg == nil then
-            table.insert(parts, 'nil')
-          else
-            table.insert(parts, format_func(arg))
-          end
-        end
-        logfile:write(table.concat(parts, '\t'), '\n')
-        logfile:flush()
-      end
+    if not open_logfile() then
+      return false
+    end
+    local info = debug.getinfo(2, 'Sl')
+    local header = string.format(
+      '[%s][%s] ...%s:%s',
+      level,
+      os.date(log_date_format),
+      info.short_src:sub(-16),
+      info.currentline
+    )
+    local parts = { header }
+    for i = 1, argc do
+      local arg = select(i, ...)
+      table.insert(parts, arg == nil and 'nil' or format_func(arg))
     end
+    assert(logfile)
+    logfile:write(table.concat(parts, '\t'), '\n')
+    logfile:flush()
   end
 end
 
--- This is put here on purpose after the loop above so that it doesn't
--- interfere with iterating the levels
-vim.tbl_add_reverse_lookup(log.levels)
+-- FIXME: DOC
+-- Should be exposed in the vim docs.
```

Comment:
It's intentionally kept private and not documented. 

See also https://github.com/neovim/neovim/pull/12150
I still wish neovim had a logging module for use in plugins, but that shouldn't be under the lsp namespace.

Reply:
Agreed; maybe after 0.10 we can sit down and discuss what we would like to offer plugins as a "standard library" and collect this in a tracking issue.

Reply:
Sorry, didn't intend to change any docs. I'll fix.

Reply:
Oh, I had also missed that this was just moved. I think we can remove the fixme